great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -541,6 +548,14 @@ def suites(self) -> SuiteFactory:
             )
         return self._suites
 
+    @property
+    def checkpoints(self) -> CheckpointFactory:
+        if not self._checkpoints:
+            raise gx_exceptions.DataContextError(
+                "DataContext requires a configured CheckpointStore to persist Checkpoints."
+            )
+        return self._checkpoints
```

Comment:
This is consistent with the suite factory but do we want to just fail fast? Can a user use GX without checkpoints? How about suites?

The only time this would actually pop up if is a user was passing their own context config into the context constructor (and purposely omitted store information).